### PR TITLE
feat(navigation): second level

### PR DESCRIPTION
**Pull request by nguyeda1**
- desktop second lvl nevigation styling
- overall navigation styling
- sidebar navigation styling
- fixed some responsivity issues

NOTE:
The sidebar is not 1to1 with the design as it would require more css "hacking" and the result would not be really worth overriding the docusaurus default behavior imho. We can iterate on this if really necessary

closes apify/apify-web#2243

—

Created via [Raycast](https://www.raycast.com?ref=signatureGithub)

### CODE REVIEW

File: apify-docs-theme/src/theme/NavbarItem/NavbarNavLink.jsx [hunk 1]
```diff
@@ -16,6 +17,9 @@ export default function NavbarNavLink({
     prependBaseUrlToHref,
     ...props
 }) {
+    const { navbar: { items = [] } } = useThemeConfig();
+    const { options: { subNavbar = [] } } = usePluginData('@apify/docs-theme');
+    const allItems = [...items, ...subNavbar];
```
Comment (barjin) [inline apify-docs-theme/src/theme/NavbarItem/NavbarNavLink.jsx:22]: `subNavbar` is an object with properties "title" and "items" - the spread will probably not work as intended

File: docusaurus.config.js [hunk 2]
```diff
@@ -22,23 +22,43 @@ module.exports = {
...
+                                {
+                                    label: 'Client for Python',
+                                    href: `${config.absoluteUrl}/client-python/`, // we need a trailing slash here, we'd get redirected there anyway
+                                    target: '_self',
+                                    rel: 'dofollow',
+                                },
+                            ],
+                        },
+                    ],
+                },
```
Comment (barjin) [inline docusaurus.config.js:61]: let's not forget to comment this out before merging :)

**Comment by B4nan** [comment]
The subnav stripe should be covering 100% of the viewport, right? also the border shoudnt be on left/right, only top/bottom

image

vs

image

it looks ok on smaller screens, appears when screen width exceeds 1646 px or sth like that

File: apify-docs-theme/src/theme/NavbarItem/NavbarNavLink.jsx (hunk 1 above)
Comment (nguyeda1) [reply apify-docs-theme/src/theme/NavbarItem/NavbarNavLink.jsx]: good catch

**Comment by nguyeda1** [comment]
> The subnav stripe should be covering 100% of the viewport, right? also the border shoudnt be on left/right, only top/bottom
> 
> image
> 
> vs
> 
> image
> 
> it looks ok on smaller screens, appears when screen width exceeds 1646 px or sth like that

Should be fixed

**Review by barjin — APPROVED** [review]

**Comment by nguyeda1** [comment]
> FYI the subnav is not present in the mobile version of the menu, we might need to swizzle some other component for that, not 100% sure now.
> 
> image
> 
> we can deal with that in another PR if you want, this problem is already present in master

@B4nan according to [this doc](https://docs-v2.apify.com/sdk-js/docs/guides/apify-platform), I think you can just add it in `sidebar.js` per each instance. Ideally, it should have the same source of truth for both `docusaurus.config.js` and `sidebar.js`

**Comment by B4nan** [comment]
This is not about sidebars, navbar navigation is not a sidebar. If we add it to sidebar, you would get a duplication on the other end (desktop version).

**Comment by nguyeda1** [comment]
> This is not about sidebars, navbar navigation is not a sidebar. If we add it to sidebar, you would get a duplication on the other end (desktop version).

@B4nan In the design the `subNav` is in the `sidebars.js` right? Or am I getting it completely wrong? I can fix the missing `navbar` items for the mobile though

SCR-20230104-ems

**Comment by B4nan** [comment]
Well, its not supposed to be in sidebar, at least not in my head, haven't seen mobile designs yet, let me have a look. It should be on the level above, when you click "back to main menu", as this is part of the main menu, not part of sidebar.

But regardless of what is in the designs, what I said still applies, sidebar is not a navbar, we cant have a subnavbar for desktop while its actually a deeply nested sidebar item - those are two different things.

**Comment by B4nan** [comment]
Ok the designs are imho wrong, I dont think we can achieve that - and it would be incorrect for the reasons above, you would add it to sidebar and it would end up in sidebar in the desktop version too. Sidebars are autogenerated based on the content, the links in second level nav are not part of sidebar on purpose, some of them cant be even there (e.g. the generated API docs in SDK and other TS repos, we have no control over that, its coming from a different plugin.

**Comment by B4nan** [comment]
This might be a bit more tricky, so I'd vote for merging it without the mobile subnav support. I can experiment with this a bit myself too, I can imagine some solutions (none of them based on the design, I'd forgot about that - it's really not feasible).

I think apart from this it should be ready to merge, or is there anything missing you are aware of? Once we merge, it should automatically reload all the other docusaurus projects.

Of course we still need to comment out the subnav as Jindra already pointed out.

**Comment by nguyeda1** [comment]
> This might be a bit more tricky, so I'd vote for merging it without the mobile subnav support. I can experiment with this a bit myself too, I can imagine some solutions (none of them based on the design, I'd forgot about that - it's really not feasible).
> 
> I think apart from this it should be ready to merge, or is there anything missing you are aware of? Once we merge, it should automatically reload all the other docusaurus projects.
> 
> Of course we still need to comment out the subnav as Jindra already pointed out.

Okay agreed then :) thx for all the insights I was not really aware of how the structure was supposed to be. I will just fix the missing mobile links for the navbar (Academy, Platform) and it is good to go

**Pull request merged**